cnd/src/swap_protocols/rfc003/alice/actions/erc20.rs

```diff
@@ -64,8 +64,8 @@ where
             actions.push(Action::Redeem(<(BL, BA)>::redeem_action(
                 HtlcParams::new_beta_params(request, response),
                 htlc_location.clone(),
-                &*self.secret_source,
-                self.secret_source.secret(),
+                &self.secret_source, // Derive identities with this.
+                self.secret_source.derive_secret(), // The secret used by Alice.
```

Comment:
To make this clearer, we could derive the secret for Alice upon instantiation of the state and do `self.secret` here? What do you think?

Reply:
I gave it a shot, it just moves the confusion to the place where we derive the secret (since at that place we now hove both the secret_source and the secret).